examples/04_pong/pong.rs

```diff
@@ -0,0 +1,241 @@
+        right: [1.0, 0.0, 0.0].into(),
+        up: [0., 1.0, 0.].into(),
+    });
+}
+
+/// Hide the cursor so it's invisible while playing.
+fn hide_cursor(world: &mut World) {
+    use amethyst::ecs::rendering::resources::WindowMessages;
+    use amethyst::winit::CursorState;
+
+    world
+        .write_resource::<WindowMessages>()
+        .send_command(|win| {
+            if let Err(err) = win.set_cursor_state(CursorState::Hide) {
+                eprintln!("Unable to make cursor hidden! Error: {:?}", err);
+            }
+        });
+}
+
+/// Initialises one paddle on the left, and one paddle on the right.
+fn initialise_paddles(world: &mut World) {
+    use {PADDLE_COLOUR, PADDLE_HEIGHT, PADDLE_VELOCITY, PADDLE_WIDTH};
+
+    let mut left_transform = LocalTransform::default();
+    let mut right_transform = LocalTransform::default();
+
+    // Correctly position the paddles.
+    left_transform.translation = [0.0, 0.0, 0.0];
+    right_transform.translation = [ARENA_WIDTH - PADDLE_WIDTH, 0.0, 0.0];
+
+    // Create the mesh and the material needed.
+    let mesh = create_mesh(
+        world,
+        generate_rectangle_vertices(0.0, 0.0, PADDLE_WIDTH, PADDLE_HEIGHT),
+    );
+
+    let material = create_colour_material(world, PADDLE_COLOUR);
+
+    // Create a left plank entity.
+    world
+        .create_entity()
+        .with(mesh.clone())
+        .with(material.clone())
+        .with(Paddle {
+            side: Side::Left,
+            width: PADDLE_WIDTH,
+            height: PADDLE_HEIGHT,
+            velocity: PADDLE_VELOCITY,
+        })
+        .with(Transform::default())
+        .with(left_transform)
+        .build();
+
+    // Create right plank entity.
+    world
+        .create_entity()
+        .with(mesh)
+        .with(material)
+        .with(Paddle {
+            side: Side::Right,
+            width: PADDLE_WIDTH,
+            height: PADDLE_HEIGHT,
+            velocity: PADDLE_VELOCITY,
+        })
+        .with(Transform::default())
+        .with(right_transform)
+        .build();
+}
+
+/// Initialises one ball in the middle-ish of the arena.
+fn initialise_balls(world: &mut World) {
+    use {ARENA_HEIGHT, ARENA_WIDTH, BALL_COLOUR, BALL_RADIUS, BALL_VELOCITY_X, BALL_VELOCITY_Y};
+
+    // Create the mesh, material and translation.
+    let mesh = create_mesh(world, generate_circle_vertices(BALL_RADIUS, 16));
+    let material = create_colour_material(world, BALL_COLOUR);
+    let mut local_transform = LocalTransform::default();
+    local_transform.translation = [ARENA_WIDTH / 2.0, ARENA_HEIGHT / 2.0, 0.0];
+
+    world
+        .create_entity()
+        .with(mesh)
+        .with(material)
+        .with(Ball {
+            radius: BALL_RADIUS,
+            velocity: [BALL_VELOCITY_X, BALL_VELOCITY_Y],
+        })
+        .with(local_transform)
+        .with(Transform::default())
+        .build();
+}
+
+
+fn load_asset<T, F>(world: &mut World, f: F) -> AssetFuture<T::Item>
```

Comment:
I heard that there is a wrapper for this in amethyst now. Might be worth looking into it ;)

Reply:
I heard [the same](https://github.com/amethyst/amethyst/pull/399#discussion_r143707137). I think it's best to wait till the two PRs are merged, and then fix all the examples at once.